util/make_distribution.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+# make_distribution.sh takes distribution artifacts in $BIN_DIR and packs them
+# into a compressed TAR archive, which is deposited into $BIN_DIR. In
+# particular, this script should be run after make_build_bin.sh.
+#
+# The actual artifacts exported are described by the $DIST_ARTIFACTS variable
+# below.
+
+. util/build_consts.sh
+
+readonly OT_VERSION="$(git describe --always)"
+echo "\$OT_VERSION = $OT_VERSION" >&2
+
+# $DIST_ARTIFACTS is a list of |find| globs to be copied out of
+# $BIN_DIR and into the OpenTitan distribution archive.
+#
+# These globs are relative to $BIN_DIR.
+readonly DIST_ARTIFACTS=(
+  'sw/*.elf'
+  'sw/*.bin'
+  'sw/*.vmem'
+  'hw/top_earlgrey/Vtop_earlgrey_verilator'
+  'hw/top_earlgrey/lowrisc_systems_top_earlgrey_nexysvideo_0.1.bit'
+)
```

Comment:
1. Why is this additional filtering necessary? It is my understanding that $BIN_DIR is the place where the distribution artifacts are stored. If the build system decides to put things there, then they should be part of the archive; the only thing that needs to be done is tar-ing this directory and adding final pieces of information to it (e.g. the README file). 

    Or is it a stopgap until the the individual build steps put their distribution artifacts into the right place? (If that's the case then please add comments.)

2. The host software isn't matched by these patterns.

Reply:
It was my understanding from our conversation last week that we wanted to centralize a place where the release is defined, so that no matter the build system, there is one source of truth for the contents of the release (rather than many, as is the case now). This can't live in Meson because Meson cannot effectively express this.

At the very least, me and Garret think that this is an important property for something as "externally"-facing as a release distribution.

Reply:
Thanks for your explanation. Looks like we didn't really have a shared understanding then. It was my understanding that $OBJDIR/dist (in https://docs.google.com/document/d/153s9kH4SyQif88K1k-6GzLCmTj1RCa4V_eWbY-hdrD4/edit) is $BIN_DIR, with  semantics as outlined in the document. That seems to be not the case. But don't worry: I'm only confused, I don't disagree.

So three points on that:
- To prevent misunderstandings it is very helpful to start with the documentation, and follow up with the implementation. In this case, it would have been helpful to update the [Google Doc](https://docs.google.com/document/d/153s9kH4SyQif88K1k-6GzLCmTj1RCa4V_eWbY-hdrD4/edit) with your implementation plan.
- In the current implementation, we effectively have three directories: $OBJ_DIR (semantics as discussed in the doc), $BIN_DIR (binaries with stable dir structure, mostly to move artifacts between build stages), and $DIST_DIR (the contents of the binary release tarball; named $OBJDIR/dist in the doc). I like that structure.
- It would be nice if the functionality currently performed by the `make_distribution.sh` script would be part of the build system, as a typical "make install" step is part of the build system. I see your point that this cannot be expressed with meson. Can you create a "pain points when using meson" document or issue and add a note on this?

Reply:
Oh. I thought you had gone and modified that design doc with what we had discussed... oops. Well, I've modified it to include everything else we need to do, and some unresolved questions. Let me know what you think, and sorry about the confusion!